lib/ansible/modules/cloud/amazon/iam_group.py

```diff
@@ -243,34 +246,37 @@ def create_or_update_group(connection, module):
     # Manage managed policies
     current_attached_policies = get_attached_policy_list(connection, module, params['GroupName'])
     if not compare_attached_group_policies(current_attached_policies, managed_policies):
+        current_attached_policies_arn_list = []
+        for policy in current_attached_policies:
+            current_attached_policies_arn_list.append(policy['PolicyArn'])
+
         # If managed_policies has a single empty element we want to remove all attached policies
         if purge_policy:
             # Detach policies not present
-            current_attached_policies_arn_list = []
```

Comment:
The reason for the restructure is that if `purge_policy` was false (basically: don't remove existing policies) we would still flag the group as changed even if no changes were being made, and would still loop through all policies that should be attached and run `attach_group_policy` on it, even if unnecessary. While my change doesn't limit that loop, it does ensure that we only return `changed: True` in cases where action would actually be taken to change things.

I'm ok simply dropping an if statement at the top of the whole block if that's what you'd prefer, but I found this discrepancy while working through testing this module and patched.